File: assignments/week3/day1/backspaceCompare.cpp
class Solution {
public:
    bool backspaceCompare(string S, string T) {
        stack<char> wS, wT;
        string wordS, wordT;
        
        for(char c: S){
            if(c != '#'){
                wS.push(c);
            }else if(!wS.empty()){
                wS.pop();
            }
        }
        
        for(char c: T){
            if(c != '#'){
                wT.push(c);
            }else if(!wT.empty()){
                wT.pop();
            }
        }
        
                
        while(!wS.empty()){
            wordS += wS.top();
            wS.pop();
        }
        
        while(!wT.empty()){
            wordT += wT.top();
            wT.pop();
        }
        
        return wordS == wordT;
    }
};

File: assignments/week3/day2/backspace_compare.py
def backspaceCompare(self, S: str, T: str) -> bool:
    stack_S = []
    stack_T = []

    for char in S:
        if char == "#" and not stack_S:
            continue
        elif char == "#":
            stack_S.pop()
        else:
            stack_S.append(char)

    for char in T:
        if char == "#" and not stack_T:
            continue
        elif char == "#":
            stack_T.pop()
        else:
            stack_T.append(char)
    return True if stack_S == stack_T else False
